python/taichi/misc/gui.py

```diff
@@ -158,6 +181,12 @@ def get_key_event(self):
         self.core.pop_key_event_head()
         return e
 
+    def peek_key_event(self):
+        e = None
```

Comment:
I'm afraid this is not peeking. Functions named `peek` usually implies idempotency, yet  inside `get_key_event()`, the event is popped.

Reply:
How about
```
get_key_event -> pop_key_event
peek_key_event -> get_key_event
```
?

Reply:
i think the current `get_key_event` is fine. Do we need peek now? If we do, maybe `get_xx()` should be implemented on top of `peek_xx()`..

Reply:
Sorry for being picky here, but i see little value of having both `pop` and `get` -- According to the current implementation, `pop_key_event()` first `wait_key_event()`, meaning that `has_key_event()` in L205 is redundant. Again, I'd suggest we just keep `get_xx()` as is, it makes the API simpler.